types/react-native/index.d.ts

```diff
@@ -5827,6 +5921,27 @@ interface PlatformWebStatic extends PlatformStatic {
     OS: 'web';
 }
 
+/**
+ * Select native platform color
+ * The color must match the string that exists on the native platform
+ *
+ * @see https://reactnative.dev/docs/platformcolor#example
+ */
+export function PlatformColor(color: string): string;
+
+type DynamicColorIOSTuple = {
+    light: string;
+    dark: string;
+};
+
+/**
+ * Specify color to display depending on the current system appearance settings
+ *
+ * @param tuple Colors you want to use for "light mode" and "dark mode"
+ * @platform ios
+ */
+export function DynamicColorIOS(tuple: DynamicColorIOSTuple): string;
```

Comment:
@tom-un, can you please review the TypeScript types for the new colors methods? I know you were investigating ways to kinda align as best as possible with what we did in Flow relative to opaque types (which don't exist in TS, right?)

Reply:
I ended up making `ColorValue` `any` for now because trying to do the actual recursive type was causing typescript errors with the current version being used. If I tried to updated it then other dependencies would break saying they specified a lower version.

I'm assuming having `any` is better than nothing as it means people can start using the new types now `0.63.0` has been released?

Reply:
**Update:** I've created a separate PR with all the `ColorValue` changes that should make it easier to review https://github.com/alexbrazier/DefinitelyTyped/pull/1

@tom-un Would you be able to take a look when you get a chance?

Reply:
Sorry I'm late to reviewing this.  I'm catching up.

I just backtracked the history and saw the three steps: first the `Opaque<K, T>` template approach, then the `Opaque<K>` approach with `K` a `unique symbol`, and then the final simplified form where the ColorValue is just `string | typeof OpaqueColorValue` with `OpaqueColorValue` a unique symbol.

I think I'm fine with the definition of the OpaqueColorValue.

But the latest iterations dropped the distinction between `ColorValue` and `ProcessedColorValue` that is in Flow.   The distinction in the Flow defs is important for internal component implementations.   I'm wondering if it would be important for an external component implementation that uses TypeScript and needs ColorValue props that could be semantic or a ProcessedColorValue.